### PR TITLE
Kafka/KsqlTopicsList cannot be deserialized

### PR DESCRIPTION
**Pull request by xvrl**
property name does not match getter

### CODE REVIEW

**Comment by xvrl** [comment]
note, I think we should enforce strict declaration of jackson properties on both getters and setters, so we don't run into those kinds of problems, it would also help make the APIs more explicit, and avoid too much magic going on.

**Comment by hjafarpour** [comment]
@xvrl did you test this with the CLI?

**Comment by xvrl** [comment]
@hjafarpour the current output of show topics returns something like that:
```
[{"kafka_topics":{"statementText":"show topics;","topics":[{"name":"MYSTREAM","registered":"false","partitionCount":2,"replicaInfo":"1","consumerCount":0,"consumerGroupCount":0}]}}]
```
which clearly shows we use `topics` as the property and not `kafka_topics`. We probably never de-serialize the value, but it would result in a null array of topics if we tried.

**Comment by xvrl** [comment]
@big-andy-coates done

**Comment by apurvam** [comment]
@xvrl can you rebase onto master and update the PR. It's impossible to review in the current form.

**Comment by xvrl** [comment]
@apurvam looks like this is a result of the force-push that was done to master, will rebase

**Comment by xvrl** [comment]
@apurvam done

File: ksql-rest-app/src/test/java/io/confluent/ksql/rest/entity/KsqlTopicsListTest.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package io.confluent.ksql.rest.entity;
```
Comment (apurvam) [inline ksql-rest-app/src/test/java/io/confluent/ksql/rest/entity/KsqlTopicsListTest.java]: No license header.

**Comment by xvrl** [comment]
@apurvam what happens is that Jackson is currently configured to allow setting fields using reflection. In this case it would pass `null` to the kafka_topics field in the constructor (since the field is not set in the json) and instead directly set the `topics` private member, since it matches the json field name.
This could lead to some really hard to track down bugs where the same variable is set twice base on the value of two different json fields, once through the constructor and once through reflection.

I added the null check to ensure the test would fail if it attempted to set the field using reflection.

**Comment by xvrl** [comment]
@apurvam the proper fix to avoid those kinds of bugs is to require jackson to be more strict and require annotations for all the fields/methods that need to be serialized / de-serialized, and reduce the amount of jackson magic that happens behind the scenes.

File: ksql-rest-app/src/test/java/io/confluent/ksql/rest/entity/KsqlTopicsListTest.java (hunk 1 above)
Comment (xvrl) [reply ksql-rest-app/src/test/java/io/confluent/ksql/rest/entity/KsqlTopicsListTest.java]: done

**Comment by xvrl** [comment]
I'll file an issue to make tighten up jackson

**Pull request merged**